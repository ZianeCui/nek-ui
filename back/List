// import { mount } from 'vue-test-utils'
// import { List } from '../../src/index'
// import { ListItem } from '../../src/index'


// // 测试slot，原理？

// describe('List.test.js', () => {
//     let cmp

//     beforeEach(() => {
//         cmp = mount(List, {
//             slots: {
//                 default: '<div class="fake-msg"></div>'
//             }
//         })
//     })

//     it('Messages are inserted in a ul.list-messages element', () => {
//         const list = cmp.find('ul.list')
//         expect(list.findAll('.fake-msg').length).toBe(1)
//     })

// })

// describe('List.test.js', () => {
//     let cmp

//     beforeEach(() => {
//         const Wrapper = {
//             render(h) {
//                 return h(ListItem, { props: { someProp: 'test' } })
//             }
//         }

//         cmp = mount(List, {
//             slots: {
//                 default: Wrapper
//             }
//         })
//     })

//     it('Messages are inserted in a ul.list-messages element', () => {
//         const list = cmp.find('ul.list')
//         expect(list.find(ListItem).isVueInstance()).toBe(true)
//     })

// })