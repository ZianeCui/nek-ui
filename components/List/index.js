import List from './src/list.vue';

List.install = function(Vue) {
  Vue.component(List.name, List);
};

export default List;