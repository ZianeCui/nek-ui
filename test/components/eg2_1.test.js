import { mount } from 'vue-test-utils'
import List from '../../components/Example/eg2/List.vue'

describe('使用mount测试组件', () => {
    let cmp

    beforeEach(() => {
        cmp = mount(List, {
            // `propsData` ??
            propsData: {
                list: ['Cat']
            }
        })
    })

    it('has received ["Cat"] as the message property', () => {
        expect(cmp.vm.list).toEqual(['Cat'])
    })

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot()
    })
})