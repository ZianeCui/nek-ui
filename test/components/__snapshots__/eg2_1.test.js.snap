// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`使用mount测试组件 has the expected html structure 1`] = `
<ul>
  <ul>
    <li
      class="list-item"
    >
      
        Cat
    
    </li>
  </ul>
</ul>
`;
